app/controllers/categories_controller.rb

```diff
@@ -0,0 +1,63 @@
+class CategoriesController < ApplicationController
+  before_action :load_category, except: %i(index new create)
+  before_action :require_admin, except: %i(index show destroy)
+
+  def index
+    @categories = Category.page(params[:page]).per(Settings.vali.cate.per)
+    @categories = Category.search(params[:name]).page(params[:name])
+  end
+
+  def new
+    @category = Category.new
+  end
+
+  def create
+    @category = Category.new category_params
+    if @category.save
+      flash[:success] = t ".create_success"
+      redirect_to @category
+    else
+      flash[:danger] = t ".create_fail"
+      render :new
+    end
+  end
+
+  def show; end
+
+  def edit; end
+
+  def update
+    if @category.update category_params
+      flash[:success] = t ".edit_success"
+      redirect_to @category
+    else
+      flash[:danger] = t ".edit_fail"
+      render :edit
+    end
+  end
+
+  def destroy
+    if @category.present?
+      @category.destroy
```

Comment:
```suggestion
    if @category&.destroy
```
Thử thế này xem ok không em

Reply:
cái này e chạy rubocop báo lỗi ạ